### ISSUE TITLE
Decompiler: raise stackalloc initializer cpblk lowerings

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#2869]
## Current state

PR #2448 made pointer and Span stackalloc initializer lowerings explicit owned
residuals. They remain `Partial`, retain `cpblk`, and end in `return default;`.
The closed issue #2428 therefore recorded the frontier but did not raise it.

Representative current fallback:

```csharp
byte* S_256 = stackalloc byte[12];
ReadOnlySpan<int> V_1 = new int[] { 1, 2, 3 };
/* Unsupported ... cpblk */
return default;
```

## Fully raised goals

```csharp
int* values = stackalloc int[] { 1, 2, 3 };
return values[0] + values[2];
```

```csharp
Span<int> values = stackalloc[] { 1, 2, 3 };
return values[0] + values[2];
```

## Required slices

1. Recognize compiler-produced `localloc` plus static-data/temporary plus
   `cpblk` initialization as one stackalloc initializer shell.
2. Recover the exact element type, count, constants, byte order, padding, and
   destination ownership for pointer and Span forms.
3. Build typed stackalloc-initializer IR and consume only the proven copy shell;
   preserve subsequent uses and evaluation order.
4. Reject non-constant sources, partial/overlapping copies, mismatched sizes,
   escaped destinations, and hand-authored `cpblk` near misses.

## Evidence

- Historical residual pin: #2448
- Closed ownership issue: #2428
- Current fixtures:
  `StackallocPointerInitializer` and `StackallocSpanInitializer` in both new
  and legacy unsafe assemblies
- Current gate requires `Partial`, `UnsupportedConstruct`, visible `cpblk`, and
  `return default;`.

Add compiler-produced positive and close-negative fixtures across SDK lowering
variants, valid C#, and compile-back opcode/control-flow evidence.

**Comment by richlander** [comment]
Implementation started on branch `raise/issue-2869-stackalloc-initializers` from `e3a88b09`.

**Comment by richlander** [comment]
Resolved by #2921 (merged), which recognizes localloc+cpblk stackalloc initializer shells and raises pointer/Span forms to a typed StackAllocArray initializer, per the required slices in this issue.
